qemu/tests/block_hotplug.py

```diff
@@ -185,14 +221,20 @@ def get_new_disk(disk1, disk2):
                                   logging.info)
             utils_test.run_virt_sub_test(test, params, env, sub_type)
 
-        for num in xrange(blk_num):
-            error_context.context("Unplug block device (iteration %d)" % iteration,
-                                  logging.info)
-            disks_before_unplug = find_disk(vm, get_disk_cmd)
-            device_list[num].unplug(vm.monitor)
-            device_list[num].verify_unplug("", vm.monitor)
-            unplug_status = utils_misc.wait_for(lambda: len(get_new_disk(find_disk(vm, get_disk_cmd),
-                                                disks_before_unplug)) != 0, pause)
+        error_context.context("Unplug block device (iteration %d)" % iteration,
+                              logging.info)
+        if controller_list:
```

Comment:
Not necessary, if `controller_list` is empty, the following `for` section will not be executed.

Don't make too much levels of indentation.